### ISSUE TITLE
Add "Max" option to Send Dialog

### ISSUE THREAD

**Issue by riongull** [dashhive/wallet-ui#41]
We really need a "Max" option on the Send screens.  

It's annoying to not be able to just send all the money out of a burner wallet (a key use case for a web wallet).
I don't like the idea of leaving dust everywhere I go by just trying to get close.

Just "Max" for now will do.  I don't know of anyone who uses 25%, 50%, etc.

I think our SDKs facilitate this somehow.  I remember dealing with it with @coolaj86 but not 100% sure we finished that feature.

**Comment by jojobyte** [comment]
@riongull please test this functionality at https://fix-dashwallet-implementation.wallet.dashincubator.dev/

**Comment by riongull** [comment]
Not working well, but I think you know that already.

**Comment by jojobyte** [comment]
@riongull please review this feature/functionality again at https://fix-dashwallet-implementation.wallet.dashincubator.dev/ and change Status to Done if you approve or back to "In Progress" if it needs more work.

**Comment by riongull** [comment]
I get this error after loading a wallet backup:

![Image](https://github.com/dashhive/wallet-ui/assets/6496310/6fe65497-9fc6-40e5-a47c-fa0f3a35d047)

As you can see, my screen is blank, so I can't test yet.

**Comment by riongull** [comment]
I tested with a new wallet (not a loaded one) and it appears to be working, but we still need to default to exact/legacy send with a checkbox for the experiemental send that sends more Dash (includes extra "stamps" - please link here to that issue if you made it somewhere).  We'll need to test that both "Max" send options work 100% reliably.

**Comment by riongull** [comment]
More testing revealed another issue:

![Image](https://github.com/dashhive/wallet-ui/assets/6496310/2051889a-fd70-4f3a-8acc-90e77de46d06)

```
Uncaught (in promise) Error: cannot spend Đ0.123024 without breaking change: add another coin of between Đ0.000002 and Đ0.001, or break change first
    at wallet._createNonTransactableError (dashwallet.js:1313:17)
    at wallet.useAllCoins (dashwallet.js:1276:22)
    at createStandardTx (wallet.js:1462:28)
    at async createTx (wallet.js:1543:13)
    at async HTMLFormElement.<anonymous> (send-or-request.js:598:45)
```

**Comment by riongull** [comment]
I was able to get past the error by sending "Half".

Presumably any value other than "Full" might have worked as well, so it does seem like an issue with the send max/full amount feature, so I moved it back to "in progress".  

Seems like this might have to be addressed by @coolaj86 at the library/sdk level.

**Comment by jojobyte** [comment]
> ```
> utils.js:1062 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')
>     at getAddressIndexFromUsage (utils.js:1062:32)
>     at getAccountWallet (wallet.js:1144:29)
>     at async main (main.js:622:14)
> getAddressIndexFromUsage	@	utils.js:1062
> getAccountWallet	@	wallet.js:1144
> await in getAccountWallet (async)		
> (anonymous)	@	main.js:1150
> ```


This appears to be related to using an older wallet backup that was created before we added `change` address support.

We likely need to add version numbers to the backups and have some form of import migration path for older backups.

**Comment by jojobyte** [comment]
> More testing revealed another issue with sending "Full":
> 
> ```
> Uncaught (in promise) Error: cannot spend Đ0.123024 without breaking change: add another coin of between Đ0.000002 and Đ0.001, or break change first
>     at wallet._createNonTransactableError (dashwallet.js:1313:17)
>     at wallet.useAllCoins (dashwallet.js:1276:22)
>     at createStandardTx (wallet.js:1462:28)
>     at async createTx (wallet.js:1543:13)
>     at async HTMLFormElement.<anonymous> (send-or-request.js:598:45)
> ```

Traced this error to https://github.com/dashhive/wallet-ui/blob/fix/dashwallet-implementation/src/helpers/wallet.js#L1464, regardless of `breakChange` being set to `true` or `false`, under certain circumstances this always fails.

@coolaj86 is there a way to manually break change before we call this function, or can the function be updated to work in the `full send` scenario.

Around this timestamp https://youtu.be/QEcXscc_r8A?t=10800 you can see the debugging of this issue.

**Comment by coolaj86** [comment]
@jojobyte If you're doing XPub Send then you get that error when sending that amount requires "unmixing" coins.

For example, you have two coins
`0.001 00000`
`0.005 02800`

And you want to send `0.001`.
Sending `0.001` costs `0.001 00193`.

Since your only other option is `0.005`, you'll have to "steal" some stamps from the `0.005`, and then you have to send change back to yourself.

When `breakChange` is set to `true`, then it will allow "unmixing" the `0.005 02800` and you'll end up with two coins:

```
0.001 01200 - to the recipient
0.005 01200 - back to yourself
``` 

Another scenario would be when you have just one coin:

```
0.005 03200
```

Here, the same rules apply and you'd end up with 2 or 3 coins, depending on the stamp values:

```
0.002 01000 - to the recipient

0.002 01000 - back to yourself
0.001 01000
```

**Comment by coolaj86** [comment]
The idea is that you should first try _without_ "breaking change", let the user check the box that says "I realize this may fingerprint my transaction, do it anyway".

In the case of a max / full send, however, you don't need to double check.

**Comment by coolaj86** [comment]
In your example it looks like you only have one coin, and it's value is

````text
0.01959386
```

So unless you were sending exactly `0.019`, it would have to break change.

Now, you've clicked "Full Send", but it only selected `0.001`, not `0.019`.

**Comment by coolaj86** [comment]
Okay, so maybe `0.01959386` was a coin I saw from another window's log.

At 3h 06m (https://youtube.com/watch?v=QEcXscc_r8A&t=11160) it looks like you only have a single coin of value `001 00000`, in which case you _can't_ send it, because all you have is dust. The other person would receive `0.000 xxxxx` which is `~= 0.000`.

**Comment by jojobyte** [comment]
> In your example it looks like you only have one coin, and it's value is
> 
> ```
> 0.01959386
> ```
> 
> So unless you were sending exactly `0.019`, it would have to break change.
> 
> Now, you've clicked "Full Send", but it only selected `0.001`, not `0.019`.

In the video, the wallet that fails has exactly `0.001`. The wallet with `0.01959385` is a different wallet and the one that is supposed to receive the funds.

**Comment by jojobyte** [comment]
Merged with #59
